lambdaclass/mirra_backend
# Stat override message

Part of https://github.com/lambdaclass/champions_of_mirra/issues/280

## Motivation

We should have a specific message for the case in which we want to override a stat's value, despite its current value.
In the immediate future this will be used to set the energy to 0 after casting the ultimate skill, in the near future it will be used to set other stats such as the health.

## Summary of changes

- Create a new StatOverride message.
- Send this message when casting a unit's ultimate skill, setting its energy to 0.

## How to test it?

You can use the SocketTester (remember to move it to the gateway directory) to simulate a battle and check that this new message is sent. You can follow these steps to do so:

```
IEx.configure(inspect: [limit: :infinity])
```

```
{:ok, user} = Champions.Users.register("Lotu")
```

```
campaign_progress = user.super_campaign_progresses |> hd()
```

```
{:ok, pid} = Gateway.SocketTester.start_link
```

```
Gateway.SocketTester.fight_level(pid, user.id, campaign_progress.level_id)
```

If you feel more comfortable using the client, start the client using [this branch](https://github.com/lambdaclass/champions_of_mirra/pull/420), go to BattleManager.cs and add a Debug.Log() in the PlayOutSteps() method, in the StatOverride case of the switch statement. Run a battle and there you should see how the frontend receives this new message when a unit is about to use the ulti.

## Checklist
